src/sardana/macroserver/macros/discretepm.py

```diff
@@ -0,0 +1,124 @@
+from sardana.macroserver.macro import Macro, Type
+import json
+import math
+
+
+class DiscretePseudoMotorConfiguration(dict):
+    def __init__(self, pseudo_obj, macro):
+        self.pseudo = pseudo_obj
+        self.macro = macro
+        _physical_motor_name = self.pseudo.physical_elements[0]
+        self.motor = macro.getMoveable(_physical_motor_name)
+        conf = self.get_configuration()
+        super(DiscretePseudoMotorConfiguration, self).__init__(conf)
+
+    def get_configuration(self):
+        return json.loads(self.pseudo.read_attribute('configuration').value)
```

Comment:
`self.pseudo` is a Taurus device. Tt is better to use taurus API to read attribute.

Reply:
I used the access to the attribute the dictionary form, however a discussion about the API still ongoing... https://github.com/taurus-org/taurus/issues/42